### PR TITLE
Init the plugin, add common parts for tests and runners, update README, add python sdk

### PR DESCRIPTION
**Pull request by nbirillo**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = "org.jetbrains.research.ml.pluginUtilities"
```
Comment (SpirinEgor) [inline build.gradle.kts:3]: `ml4se`?)
But I think we don't have concrete artifacts. For example, I use only `org.jetbrains.research`.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+version = "1.0"
+
+fun properties(key: String) = project.findProperty(key).toString()
+
+plugins {
+    java
+    kotlin("jvm") version "1.5.10" apply true
```
Comment (SpirinEgor) [inline build.gradle.kts:10]: `1.5.21` is the latest version, why don't use it?

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    id("org.jetbrains.intellij") version "1.0" apply true
```
Comment (SpirinEgor) [inline build.gradle.kts:11]: Same here, why you set a little bit outdated versions?

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        plugin("org.jlleitschuh.gradle.ktlint")
+    }
+
+    repositories {
+        mavenCentral()
+        jcenter()
+    }
+
+    dependencies {
+        implementation(kotlin("stdlib-jdk8"))
```
Comment (SpirinEgor) [inline build.gradle.kts:31]: Don't hate me for that version definition review :) But you set JDK 11, why do you specify JDK-8 here?

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8-bin.zip
```
Comment (SpirinEgor) [inline gradle/wrapper/gradle-wrapper.properties:3]: Gradle 7?)

File: plugin-utilities-plugin/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    // Input directory with files
+    @get:Input
+    val input: String? by project
+
+    // Output directory to store indexes and methods data
+    @get:Input
+    val output: String? by project
+
+    init {
+        jvmArgs = listOf("-Djava.awt.headless=true", "--add-exports", "java.base/jdk.internal.vm=ALL-UNNAMED")
```
Comment (SpirinEgor) [inline plugin-utilities-plugin/build.gradle.kts:23]: I also add `"-Djdk.module.illegalAccess.silent=true"` to supress warnings

File: plugin-utilities-test/src/main/kotlin/org/jetbrains/research/pluginUtilities/util/FileTestUtil.kt [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+import com.intellij.psi.PsiFile
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+import java.io.File
+
+enum class Type {
+    Input, Output
+}
+
+class TestFileFormat(private val prefix: String, private val extension: Extension, val type: Type) {
```
Comment (Furetur) [inline plugin-utilities-test/src/main/kotlin/org/jetbrains/research/pluginUtilities/util/FileTestUtil.kt:11]: This assumes that all tested operations work with input and output files (not directories). In my case, preprocessing is done for the entire project, it has an input directory and output directory. This abstraction cannot be used in my case if I understand everything correctly.

File: plugin-utilities-test/src/test/kotlin/org/jetbrains/research/pluginUtilities/util/SimpleParametrizedTest.kt [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    @Parameterized.Parameter(0)
+    var inFile: File? = null
+
+    @JvmField
+    @Parameterized.Parameter(1)
+    var outFile: File? = null
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{index}: ({0}, {1})")
```
Comment (Furetur) [inline plugin-utilities-test/src/test/kotlin/org/jetbrains/research/pluginUtilities/util/SimpleParametrizedTest.kt:22]: Can `ParametrizedBaseTest` be used in a situation where there's no `outFile`? For example, when we test if project opens we only have the `inFile` -- path to project directory. We open the project and calculate our metric

File: plugin-utilities-test/src/main/kotlin/org/jetbrains/research/pluginUtilities/util/FileTestUtil.kt (hunk 7 above)
Comment (nbirillo) [reply plugin-utilities-test/src/main/kotlin/org/jetbrains/research/pluginUtilities/util/FileTestUtil.kt:11]: Yes, thank you! I will fix it today to add an able of using folders as input

File: plugin-utilities-test/src/test/kotlin/org/jetbrains/research/pluginUtilities/util/SimpleParametrizedTest.kt (hunk 8 above)
Comment (nbirillo) [reply plugin-utilities-test/src/test/kotlin/org/jetbrains/research/pluginUtilities/util/SimpleParametrizedTest.kt:22]: Yes, you can

Comment (nbirillo) [reply plugin-utilities-test/src/test/kotlin/org/jetbrains/research/pluginUtilities/util/SimpleParametrizedTest.kt]: I added an example for this case

**Review by Furetur — APPROVED** [review]

File: plugin-utilities-plugin/build.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,44 @@
+group = rootProject.group
+version = rootProject.version
+
+dependencies {
+    implementation("com.xenomachina:kotlin-argparser:2.0.7")
+}
+
+// TODO: move into the project
+open class CliTask : org.jetbrains.intellij.tasks.RunIdeTask() {
```
Comment (Furetur) [inline plugin-utilities-plugin/build.gradle.kts:9]: I don't think that this code can be easily reused when creating CLI runners that accept extra arguments. In my case, `PreprocessingRunner` should accept path to Android SDK and I do not see how I can effectively reuse this code. If I extend this class I will still have the `runner` field which I do not need. Also, my runner will require a new entrypoint like `./gradlew -q preprocess`

Comment (nbirillo) [reply plugin-utilities-plugin/build.gradle.kts]: We discussed it in DM. Should we resolve it or not?

**Pull request merged**